Mathlib/Combinatorics/SetFamily/Compression/UV.lean

```diff
@@ -0,0 +1,293 @@
+#align uv.compress_of_disjoint_of_le UV.compress_of_disjoint_of_le
+
+/-- `a` is in the UV-compressed family iff it's in the original and its compression is in the
+original, or it's not in the original but it's the compression of something in the original. -/
+theorem mem_compression :
+    a ∈ 𝓒 u v s ↔
+      a ∈ s ∧ compress u v a ∈ s ∨ a ∉ s ∧ ∃ b ∈ s, compress u v b = a := by
+  simp [compression, mem_union, mem_filter, mem_image, and_comm]
+#align uv.mem_compression UV.mem_compression
+
+@[simp]
+theorem compress_self (u a : α) : compress u u a = a := by
+  unfold compress
+  split_ifs
+  · exact ‹Disjoint u a ∧ u ≤ a›.1.symm.sup_sdiff_cancel_right
+  · rfl
+#align uv.compress_self UV.compress_self
+
+@[simp]
+theorem compression_self (u : α) (s : Finset α) : 𝓒 u u s = s :=
+  by
+  unfold compression
+  convert union_empty s
+  · ext a
+    simp [mem_filter, compress_self, and_self_iff]
+  · refine' eq_empty_of_forall_not_mem fun a ha => _
+    simp_rw [mem_filter, mem_image, compress_self] at ha
+    obtain ⟨⟨b, hb, rfl⟩, hb'⟩ := ha
+    -- Porting note: Previous code was: exact hb' hb
+    simp at hb'; exact hb' hb
+#align uv.compression_self UV.compression_self
+
+/-- Any family is compressed along two identical elements. -/
+theorem is_compressed_self (u : α) (s : Finset α) : IsCompressed u u s :=
+  compression_self u s
+#align uv.is_compressed_self UV.is_compressed_self
+
+theorem compress_disjoint (u v : α) :
+    Disjoint (s.filter fun a => compress u v a ∈ s)
+      ((s.image <| compress u v).filter fun a => a ∉ s) :=
+  disjoint_left.2 fun a ha₁ ha₂ =>
+  -- Porting note: Previous code was:
+  -- (mem_filter.1 ha₂).2 (mem_filter.1 ha₁).1
+  by simp_all
+#align uv.compress_disjoint UV.compress_disjoint
+
+/-- Compressing an element is idempotent. -/
+@[simp]
+theorem compress_idem (u v a : α) : compress u v (compress u v a) = compress u v a :=
+  by
+  unfold compress
+  split_ifs with h h' <;> try rfl
+  rw [le_sdiff_iff.1 h'.2, sdiff_bot, sdiff_bot, sup_assoc, sup_idem]
+#align uv.compress_idem UV.compress_idem
+
+theorem compress_mem_compression (ha : a ∈ s) : compress u v a ∈ 𝓒 u v s := by
+  rw [mem_compression]
+  by_cases compress u v a ∈ s
+  · rw [compress_idem]
+    exact Or.inl ⟨h, h⟩
+  · exact Or.inr ⟨h, a, ha, rfl⟩
+#align uv.compress_mem_compression UV.compress_mem_compression
+
+-- This is a special case of `compress_mem_compression` once we have `compression_idem`.
+theorem compress_mem_compression_of_mem_compression (ha : a ∈ 𝓒 u v s) :
+    compress u v a ∈ compression u v s := by
+  rw [mem_compression] at ha⊢
+  simp only [compress_idem, exists_prop]
+  obtain ⟨_, ha⟩ | ⟨_, b, hb, rfl⟩ := ha
+  · exact Or.inl ⟨ha, ha⟩
+  · exact Or.inr ⟨by rwa [compress_idem], b, hb, (compress_idem _ _ _).symm⟩
+#align
+  uv.compress_mem_compression_of_mem_compression
+  UV.compress_mem_compression_of_mem_compression
+
+-- Porting note: Due to `Bool`s, this proof needed rewriting. Could use severe golfing
```

Comment:
That makes me think it's not even worth porting this file before we fix the `Bool`-valued functions 😞